Sources/StreamChat/StateLayer/Coordinators/ChannelListQueryLinkingCoordinator.swift

```diff
@@ -0,0 +1,100 @@
+//
+// Copyright © 2024 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+/// Listens to local events and adds or removes channels from ``ChannelListQueryDTO.channels``.
+@available(iOS 13.0, *)
+struct ChannelListQueryLinkingCoordinator {
+    private let eventsController: EventsController
+    private let eventsDelegate: EventsDelegate
+    
+    init(state: ChatListState, query: ChannelListQuery, dynamicFilter: ((ChatChannel) -> Bool)?, channelListUpdater: ChannelListUpdater, eventsController: EventsController, database: DatabaseContainer, clientConfig: ChatClientConfig) {
+        self.eventsController = eventsController
+        eventsDelegate = EventsDelegate(database: database)
+        
+        eventsController.delegate = eventsDelegate
+        eventsDelegate.linkingHandler = { [weak state] channel, shouldLink in
```

Comment:
```suggestion
        eventsDelegate.linkingHandler = { [weak state] channel, **shouldLink** in
```
Where does the shouldLink come from?